src/robotFactory.js

```diff
@@ -1,10 +1,91 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, position, chipVersion) {
+    const { x = 0, y = 0 } = { ...position };
 
-class FlyingRobot {}
+    this.name = name;
+    this.weight = weight;
 
-class DeliveryDrone {}
+    this.coords = {
+      x: x,
+      y: y,
+    };
+
+    this.chipVersion = chipVersion;
+  }
+
+  goForward(step = 1) {
+    this.coords.y += step;
+  }
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+  }
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  }
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  }
+
+  getInfo() {
+    const nameInfo = `Robot: ${this.name}, `;
+    const chipInfo = `Chip version: ${this.chipVersion}, `;
+    const weightInfo = `Weight: ${this.weight}`;
+
+    return nameInfo + chipInfo + weightInfo;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, position, chipVersion) {
+    super(name, weight, position, chipVersion);
+
+    const { x = 0, y = 0, z = 0 } = position;
+
+    this.coords = {
+      x: x,
+      y: y,
+      z: z,
```

Comment:
should work with this shortcut
```suggestion
      x,
      y,
      z,
```

Reply:
so as I removed destructuring assignment, now it looks like
`    this.coords = {
      x: position.x || 0,
      y: position.y || 0,
      z: position.z || 0,
    };
`